rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -2987,6 +3020,87 @@ extern "C" rmw_ret_t rmw_get_node_names(
   return RMW_RET_BAD_ALLOC;
 }
 
+extern "C" rmw_ret_t rmw_get_node_names_with_security_contexts(
+  const rmw_node_t * node,
+  rcutils_string_array_t * node_names,
+  rcutils_string_array_t * node_namespaces,
+  rcutils_string_array_t * security_contexts)
+{
+  RET_WRONG_IMPLID(node);
+  auto node_impl = static_cast<CddsNode *>(node->data);
+  if (rmw_check_zero_rmw_string_array(node_names) != RMW_RET_OK ||
+    rmw_check_zero_rmw_string_array(node_namespaces) != RMW_RET_OK)
```

Comment:
@ivanpauno should the same check be performed on `security_contexts` if not `nullptr`?

Reply:
It seems to me that you're looking to the wrong function:
https://github.com/ros2/rmw_cyclonedds/blob/ad4879663c2244ff88604292b239cc0400bf94df/rmw_cyclonedds_cpp/src/rmw_node.cpp#L3093-L3100

Reply:
I think not. The snippet you refer to takes care of finalizing it if present, but I don't see a `rmw_check_zero_rmw_string_array` counterpart and [right below](https://github.com/ros2/rmw_cyclonedds/blob/ad4879663c2244ff88604292b239cc0400bf94df/rmw_cyclonedds_cpp/src/rmw_node.cpp#L3057) what is referred to by `security_context` is initialized without checking whether it was `nullptr` or not.

Reply:
See https://github.com/ros2/rmw_cyclonedds/pull/106/commits/1c072f236b94853a1042b468215625726d5c8011.